projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -407,45 +407,65 @@ public function is_task_id_complete( $task_id ) {
 	 *
 	 * @param Task_List $task_list Task List.
 	 *
-	 * @return bool True if valid, false if not.
+	 * @return null|WP_Error Null if valid, WP_Error if not.
 	 */
 	public static function validate_task_list( $task_list ) {
+		$error_code     = 'validate_task_list';
+		$error_messages = array();
+
 		if ( ! is_array( $task_list ) ) {
-			return false;
+			// If we don't have an array, skip later validation.
+			return new WP_Error( $error_code, 'Invalid task list' );
 		}
 
 		if ( ! isset( $task_list['id'] ) ) {
-			_doing_it_wrong( 'validate_task_list', 'The Launchpad task list being registered requires a "id" attribute', '6.1' );
-			return false;
+			$msg = 'The Launchpad task list being registered requires a "id" attribute';
+			_doing_it_wrong( 'validate_task_list', esc_html( $msg ), '6.1' );
```

Comment:
Do we need to keep _doing_it_wrong since we're now returning a WP_Error? I admit, I read over the function docs and I'm not entirely sure what it signifies 😆

Reply:
Yes! We decided [here](https://github.com/Automattic/wp-calypso/issues/77939#issuecomment-1631136186) to keep logging it with `_doing_it_wrong` for now.